src/components/AlertModal/AlertModal.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { FunctionComponent, useMemo } from "react";
+import { Modal, StyleSheet, Text, View } from "react-native";
+import AlertLogo from "../../../assets/icons/alert.svg";
+import { DarkButton } from "../Layout/Buttons/DarkButton";
+import { SecondaryButton } from "../Layout/Buttons/SecondaryButton";
+import { DangerButton } from "../Layout/Buttons/DangerButton";
+import { size } from "../../common/styles";
+
+const styles = StyleSheet.create({
+  centeredView: {
+    flex: 1,
+    width: "100%",
+    justifyContent: "center",
+    alignItems: "center",
+    alignSelf: "center",
+    backgroundColor: "rgba(0,0,0,0.6)"
+  },
+  modalView: {
+    marginBottom: 0,
+    paddingTop: 24,
+    width: 280,
+    backgroundColor: "white",
+    borderRadius: size(1),
+    shadowColor: "#000",
+    shadowOffset: {
+      width: 0,
+      height: 2
+    },
+    shadowOpacity: 0.25,
+    shadowRadius: 3.84,
+    elevation: 5
+  },
+  alertIcon: {
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "center",
+    alignSelf: "center",
+    marginBottom: 9
+  },
+  modalTitle: {
+    fontWeight: "bold",
+    fontFamily: "brand-bold",
+    fontSize: 20,
+    color: "#305367",
+    textAlign: "center",
+    marginHorizontal: 28,
+    marginBottom: 4
+  },
+  modalText: {
+    marginHorizontal: 32.5,
+    marginBottom: 32,
+    textAlign: "center",
+    fontSize: 16,
+    color: "#305367"
+  },
+  modalGroupButton: {
+    flexDirection: "row",
+    marginHorizontal: 16,
+    marginBottom: 24
+  },
+  modalSecondaryBtm: {
+    marginRight: 8,
+    flexShrink: 1
+  },
+  modalPrimaryButton: {
+    flex: 1
+  }
+});
+type AlertType = "ERROR" | "WARN" | "CONFIRM" | "INFO";
+
+// Add on the collection of action button texts
+// try to declare object as { [key:string]: CallToActionButtonTexts }
+// but somehow CallToActionKey unable to determine the keys on compile time
+const callToActionCollection = Object.freeze({
+  YES_NO: { primaryActionText: "YES", secondaryActionText: "NO" },
+  OK_CANCEL: { primaryActionText: "OK", secondaryActionText: "CANCEL" },
+  CONFIRM_CANCEL: {
+    primaryActionText: "CONFIRM",
+    secondaryActionText: "CANCEL"
+  }
+});
+
+interface CallToActionButtonTexts {
+  primaryActionText: string;
+  secondaryActionText?: string;
+}
+
+export type CallToActionKey = keyof typeof callToActionCollection;
```

Comment:
Interesting!

What's the reason for deriving the `CallToActionKey` from the implementation, rather than the inverse (i.e. define the `CallToActionKey`, then `callToActionCollection` adheres to the type by ensuring those keys are defined? I'm thinking we probably want then for different languages, what do you think?

Reply:
Maybe you enlighten me on this. I have tried to declare the object by using the following
```ts
const callToActionCollection: { [key:"YES_NO" | "OK_CANCEL" | "CONFIRM_CANCEL"]: CallToActionButtonTexts }
```
But the `key` was expected to be `string` or `number`, not string literals.

Alternatively is to make use of `Map<"YES_NO" | "OK_CANCEL" | "CONFIRM_CANCEL" , CallToActionButtonTexts>`

How do you intend to cater for different languages?

Reply:
Just tried this approach:
```
type CallToActionKey = "YES_NO" | "OK_CANCEL" | "CONFIRM_CANCEL";
type CallToActionButtonTexts = {
  primaryActionText: string;
  secondaryActionText?: string;
};
type CallToActionCollection = {
  [key in CallToActionKey]: CallToActionButtonTexts;
};

const callToActionCollection: CallToActionCollection = {
  YES_NO: { primaryActionText: "YES", secondaryActionText: "NO" },
  OK_CANCEL: { primaryActionText: "OK", secondaryActionText: "CANCEL" },
  CONFIRM_CANCEL: {
    primaryActionText: "CONFIRM",
    secondaryActionText: "CANCEL"
  }
};
```

For the different languages, I'm thinking they can be supplied by the campaign configuration context. Then `AlertModal` can directly access the context to find the appropriate mappings for `callToActionCollection`. Similar to a `ThemeProvider` in CSS-in-JS libraries.

Reply:
nice! TIL.
Updated and pushed